src/c_notification.cpp

```diff
@@ -0,0 +1,80 @@
+
+#include <stdexcept>
+#include <string>
+
+#include "lib/units.h"
+
+#include "c_notification.h"
+
+namespace spacegun {
+  using std::string;
+  using aronnax::Color;
+  using aronnax::Component;
+  using aronnax::Vector2d;
+
+  extern const string COMPONENT_TYPE_NOTIFICATION;
+
+  void Notification::setFontSize(int size)
+  {
+    fontSize_ = size;
+  }
+
+  void Notification::setColor(const Color& color)
+  {
+    color_ = color;
+    for (auto tl : lines_) {
+      tl.col = color;
+    }
+  }
+
+  Color Notification::getColor()
+  {
+    return color_;
+  }
+
+  int Notification::addLine(string msg)
+  {
+    auto newLine = createNewLine(msg);
+    lines_.push_back(newLine);
+    return lines_.size() - 1;
+  }
+
+  string Notification::getLine(int lineNum)
+  {
+    return lines_[lineNum].msg;
+  }
+
+  void Notification::updateLine(int lineNum, string msg)
+  {
+    if (lineNum == -1) {
```

Comment:
This could improve by also checking that the element of the index exists in the vector.